.github/assert-deployed-bytecode.js

```diff
@@ -0,0 +1,104 @@
+const { AssertionError } = require('chai')
+const chalk = require('chalk')
+const { web3 } = require('hardhat')
+const { readJSON } = require('../scripts/helpers/fs')
+const { APPS_TO_NAMES, CONTRACTS_TO_NAMES, IGNORE_METADATA_CONTRACTS } = require('./deployed-bytecode-consts')
+
+function compareSymbolWise(first, second, ignoreMetadata) {
+  let compareEnd = first.length
+  if (ignoreMetadata) {
+    compareEnd = first.indexOf('a264')
```

Comment:
It will work for the particular case. But as you can see here https://docs.soliditylang.org/en/v0.6.12/metadata.html#encoding-of-the-metadata-hash-in-the-bytecode there can be additional fields there.
BTW how you cont the actual length? I can't figure out

Reply:
I see where are you coming from... I was incorrect initialy about constant offset calculation, indeed.

Though, we still should avoid using `a264`:
>The CBOR mapping can also contain other keys, so it is better to fully decode the data instead of relying on it starting with 0xa264.

I also checked out docs for the [`solc 0.4.24`](https://docs.soliditylang.org/en/v0.4.24/metadata.html), and there is no `0xa264` at all. 

---

The game-changer is the following:
>Since the mapping might contain more keys (see below) and the beginning of that encoding is not easy to find, its length is added in a two-byte big-endian encoding. 

Looks like the best option here is to use the last two bytes encoded metadata length!
We could read them and now we know the exact offset to skip if needed.

I have checked and it works:

* Solidity 0.6.12 docs sample:
```
0xa2
0x64 'i' 'p' 'f' 's' 0x58 0x22 <34 bytes IPFS hash>
0x64 's' 'o' 'l' 'c' 0x43 <3 byte version encoding>
0x00 0x33
```
here:
`0x00 0x33` == 51
`length(0xa2...<3 byte version encoding>)` == 51

* Solidity 0.4.24 docs sample:
```
0xa1 0x65 'b' 'z' 'z' 'r' '0' 0x58 0x20 <32 bytes swarm hash> 0x00 0x29
```
here:
`0x00 0x29` == 41
`length(0xa1...<32 bytes swarm hash>` == 41